Documentation/edgefs-teardown.md

```diff
@@ -0,0 +1,86 @@
+
+Consider to use `resurrect` and `zapDevices` flags if you planning to teardown cluster but keep current device configuration intact.
+
+## Delete the NFS, S3 and iSCSI artifacts
+First you will need to clean up the resources created on top of the Rook cluster. This includes any pods and applications consuming CSI provisioned storage as well as respected protocol CRDs.
+
+## Delete the Cluster CRD
+After those protocol resources have been cleaned up, you can then delete your Rook cluster. This is important to delete **before removing the Rook operator or else resources may not be cleaned up properly**.
+```console
+kubectl -n rook-edgefs delete cluster.edgefs.rook.io rook-edgefs
+```
+
+Verify that the cluster CRD has been deleted before continuing to the next step.
+```
+kubectl -n rook-edgefs get cluster.edgefs.rook.io
+```
+
+## Delete the Operator
+This will begin the process of all cluster resources being cleaned up, after which you can delete the operator and related resources such as the agent and discover daemonsets with the following:
+```console
+kubectl delete -f operator.yaml
+```
+
+Optionally remove the rook-edgefs namespace if it is not in use by any other resources.
+```
+kubectl delete namespace rook-edgefs
+```
+
+## Delete the data on hosts
+IMPORTANT: The final cleanup step requires deleting files on each host in the cluster. All files under the `dataDirHostPath` property specified in the cluster CRD will need to be deleted. Otherwise, inconsistent state will remain when a new cluster is started. Consider to use `resurrect` and `zapDevices` flags if you planning to teardown cluster but keep current device configuration intact.
+
+Connect to each machine and delete `/var/lib/rook`, or the path specified by the `dataDirHostPath`.
+
+In the future this step will not be necessary when we build on the K8s local storage feature.
+
+If you modified the demo settings, additional cleanup is up to you for devices, host paths, etc.
+
+## Troubleshooting
+If the cleanup instructions are not executed in the order above, or you otherwise have difficulty cleaning up the cluster, here are a few things to try.
+
+The most common issue cleaning up the cluster is that the `rook` namespace or the cluster CRD remain indefinitely in the `terminating` state. A namespace cannot be removed until all of its resources are removed, so look at which resources are pending termination.
+
+Look at the pods:
+```
+kubectl -n rook-edgefs get pod
+```
+If a pod is still terminating, you will need to wait or else attempt to forcefully terminate it (`kubectl delete pod <name>`).
+
+Now look at the cluster CRD:
+```
+kubectl -n rook-edgefs get cluster.edgefs.rook.io
+```
+If the cluster CRD still exists even though you have executed the delete command earlier, see the next section on removing the finalizer.
+
+### Removing the Cluster CRD Finalizer
```

Comment:
have you all seen this lingering finalizer happening for EdgeFS also during your testing?